lib/pmdtester/builders/html_report_builder.rb

```diff
@@ -122,53 +118,54 @@ def build_violations_section(doc, violation_diffs)
 
     def build_violation_table(doc, key, value)
       doc.table(class: 'bodyTable', border: '0') do
-        doc.thead do
-          doc.tr do
-            doc.th
-            doc.th 'priority'
-            doc.th 'Rule'
-            doc.th 'Message'
-            doc.th 'Line'
-          end
+        build_violation_table_head(doc)
+        build_violation_table_body(doc, key, value)
+      end
+    end
+
+    def build_violation_table_head(doc)
+      doc.thead do
+        doc.tr do
+          doc.th
+          doc.th 'priority'
+          doc.th 'Rule'
+          doc.th 'Message'
+          doc.th 'Line'
         end
+      end
+    end
 
-        doc.tbody do
-          a_index = 1
-          value.each do |pmd_violation|
-            doc.tr(class: pmd_violation.branch == 'base' ? 'a' : 'b') do
-              # The anchor
-              doc.td do
-                doc.a(id: "A#{a_index}", href: "#A#{a_index}") do
-                  doc.text '#'
-                end
-              end
+    def build_violation_table_body(doc, key, value)
+      doc.tbody do
+        a_index = 1
+        value.each do |pmd_violation|
+          doc.tr(class: pmd_violation.branch == 'base' ? 'a' : 'b') do
```

Comment:
How does it change, if you factor out this block into a own method "build_violation_table_row(a_index, pmd_violation)"?